library/network/src/modules/SuSEFirewall2Services.rb

```diff
@@ -0,0 +1,608 @@
+# you may find current contact information at www.novell.com
+#
+# ***************************************************************************
+#
+# File:	modules/SuSEFirewallServices.rb
+# Package:	Firewall Services, Ports Aliases.
+# Summary:	Definition of Supported Firewall Services and Port Aliases.
+# Authors:	Lukas Ocilka <locilka@suse.cz>
+#
+# Global Definition of Firewall Services
+# Defined using TCP, UDP and RPC ports and IP protocols and Broadcast UDP
+# ports. Results are cached, so repeating requests are answered faster.
+
+require "yast"
+require_relative "SuSEFirewallServices"
+
+module Yast
+  class SuSEFirewall2ServicesClass < SuSEFirewallServicesClass
+    include Yast::Logger
+
+    # this is how services defined by package are distinguished
+    DEFINED_BY_PKG_PREFIX = "service:"
+
+    SERVICES_DIR = "/etc/sysconfig/SuSEfirewall2.d/services/"
+
+    # please, check it with configuration in refresh-srv-def-by-pkgs-trans.sh script
+    SERVICES_TEXTDOMAIN = "firewall-services"
+
+    READ_ONLY_SERVICE_FEATURES = ["name", "description"]
+
+    IGNORED_SERVICES = ["TEMPLATE", "..", "."]
+
+    TEMPLATE_SERVICE_NAME = "template service"
+    TEMPLATE_SERVICE_DESCRIPTION = "opens ports for foo in order to allow bar"
+
+    def main
+      textdomain "base"
+
+      Yast.import "FileUtils"
+
+      #
+      # IF YOU NEED TO ADD ANOTHER SERVICE, CREATE THE SERVICE DEFINITION
+      # IN A FILE AND ADD IT TO THE PACKAGE TO WHICH IT BELONGS.
+      # USE /etc/sysconfig/SuSEfirewall2.d/services/TEMPLATE FOR THAT.
+      #
+      # MORE INFORMATION IN FEATURE #300687: Ports for SuSEfirewall added via packages.
+      # ANOTHER REFERENCE: Bugzilla #246911.
+      #
+      # See also http://kobliha-suse.blogspot.cz/2008/06/firewall-services-defined-by-packages.html
+      #
+      #
+      # Names assigned to Port and Protocol numbers can be found
+      # here:
+      #
+      # http://www.iana.org/assignments/protocol-numbers
+      # http://www.iana.org/assignments/port-numbers
+      #
+      # Format of SERVICES
+      #
+      #   "service-id" : $[
+      #     "name"            : _("Service Name"),
+      #     "tcp_ports"       : list <tcp_ports>,
+      #     "udp_ports"       : list <udp_ports>,
+      #     "rpc_ports"       : list <rpc_ports>,
+      #     "ip_protocols"    : list <ip_protocols>,
+      #     "broadcast_ports" : list <broadcast_ports>,
+      #   ],
+      #
+      @services = nil
+
+      # firewall needs restarting
+      @sfws_modified = false
+
+      @known_services_features = {
+        "TCP"       => "tcp_ports",
+        "UDP"       => "udp_ports",
+        "RPC"       => "rpc_ports",
+        "IP"        => "ip_protocols",
+        "BROADCAST" => "broadcast_ports"
+      }
+
+      @known_metadata = { "Name" => "name", "Description" => "description" }
+
+      # Services definitions for conversion to the new ones.
+      @OLD_SERVICES = {
```

Comment:
This is part of the old SF2 code. However, with all these splits I would prefer it for code fixes to be made in a separate commit because it is too painful to rebase, do code fixes and then get conflicts in the split files.